pycheribuild/projects/cross/freertos.py

```diff
@@ -76,22 +94,27 @@ def __init__(self, config: CheriConfig):
         # Override that with no and set the appopriate flags here.
         self.make_args.set(NIX_ENV="no")
 
-        # Only build 64-bit FreeRTOS as cheribuild currently only supports building
-        # for RV64
-        self.make_args.set(RISCV_XLEN="64")
-
         # Set sysroot Makefile arg to pick up libc
         self.make_args.set(SYSROOT=str(self.sdk_sysroot))
 
+        # Is purecap?
+        self.is_purecap = self.target_info.target.is_cheri_purecap()
+
         # Add compiler-rt location to the search path
         # self.make_args.set(LDFLAGS="-L"+str(self.compiler_resource / "lib"))
 
-        if self.target_info.target.is_cheri_purecap():
-            # CHERI-RISC-V sophisticated Demo with more advanced device drivers
-            # and currently only runs on FPGA-GFE, purecap
-            self.supported_freertos_demos.append("RISC-V_Galois_P1")
-            self.supported_demo_apps["RISC-V_Galois_P1"] = ["main_blinky", "main_netboot"]
+        if self.demo == "RISC-V_Galois_P1":
+            if self.crosscompile_target.is_riscv32(include_purecap=self.is_purecap):
```

Comment:
Setting include_purecap based on the current target info makes no sense, and it's also really weird to have XARCH=64 in an else branch that could be hit for riscv32-purecap if self.is_purecap is False (which can't happen, because it's all redundantly defined, but is just another reason why this is nonsense).

Reply:
Besides, don't you want to give an error if someone's trying to build this demo as non-purecap given that's not supported by our fork?